org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -413,7 +419,10 @@ public CompletionItem toCompletionItem(CompletionProposal proposal, int index) {
 		if ($.getTextEdit() != null) {
 			String newText = $.getTextEdit().isLeft() ? $.getTextEdit().getLeft().getNewText() : $.getTextEdit().getRight().getNewText();
 			Range range = $.getTextEdit().isLeft() ? $.getTextEdit().getLeft().getRange() : ($.getTextEdit().getRight().getInsert() != null ? $.getTextEdit().getRight().getInsert() : $.getTextEdit().getRight().getReplace());
-			if (range != null && newText != null) {
+			String filterText = CompletionProposalDescriptionProvider.createMethodProposalDescription(proposal).toString();
+			if (range != null && !filterText.isEmpty() && $.getKind() == CompletionItemKind.Method) {
```

Comment:
I think you'd also want to include `CompletionItemKind.Constructor` . Those aren't included in the `Method` kind.